### PR TITLE
spawning a new child should not inherit sig handlers

### PR DESCRIPTION
**Pull request by paulcallen**
Signed-off-by: Paul Allen <paul.c.allen@microsoft.com>

### CODE REVIEW

File: kernel/exec.c [hunk 1]
```diff
@@ -849,6 +850,9 @@ int myst_exec(
 
     process = thread->process;
 
+    /* reset signactions */
```
Comment (vtikoo) [inline kernel/exec.c:853]:
```suggestion
    /* reset sigactions */
```

File: tests/spawn/child-signal.c [hunk 2]
```diff
@@ -41,13 +41,25 @@ int sigtest1()
...
+    return 0;
+}
+
 int main(int argc, const char* argv[], const char* envp[])
 {
     if ((argc = 2) && (strcmp(argv[1], "sigtest1") == 0))
     {
         assert(sigtest1() == 0);
     }
+    else if ((argc = 2) && (strcmp(argv[1], "clear_sighandler") == 0))
```
Comment (vtikoo) [inline tests/spawn/child-signal.c:57]:
```suggestion
    else if ((argc == 2) && (strcmp(argv[1], "clear_sighandler") == 0))
```

**Review by vtikoo — APPROVED** [review]
LGTM, apart from minor issue in tests/spawn.

File: kernel/signal.c [hunk 3]
```diff
@@ -68,10 +68,17 @@ int myst_signal_init(myst_process_t* process)
 {
     int ret = 0;
 
-    process->signal.sigactions = calloc(NSIG, sizeof(posix_sigaction_t));
+    if (!process->signal.sigactions)
```
[the remark this answers is not in the document]
Comment (paulcallen) [reply kernel/signal.c:71]: A forked process has these, but once we do an exec on that process we need to wipe out the existing handlers.
This is primarily going to exist when called from exec

[the remark this answers is not in the document]
Comment (vtikoo) [reply kernel/signal.c:71]: Ah okay. So `myst_signal_init` gets called twice, once during fork and once during exec. The exec one is clearing the sigactions copied during fork.

[the remark this answers is not in the document]
Comment (vtikoo) [reply kernel/signal.c:71]: A comment might be helpful here.

File: tests/spawn/child-signal.c (hunk 2 above)
Comment (paulcallen) [reply tests/spawn/child-signal.c:57]: yikes!

File: kernel/signal.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (paulcallen) [reply kernel/signal.c:71]: main process and child exec process

[the remark this answers is not in the document]
Comment (paulcallen) [reply kernel/signal.c:71]: ok. this is also called from VM clone (fork) too. I am going to look at this again as fork does inherit the handlers

[the remark this answers is not in the document]
Comment (paulcallen) [reply kernel/signal.c:79]: its all good. the clone instantly copies the handlers over from the parent process.
comment added

**Pull request merged**